store/postgres/src/deployment_store.rs

```diff
@@ -87,7 +87,7 @@ pub struct StoreInner {
     conn_round_robin_counter: AtomicUsize,
 
     /// A cache of commonly needed data about a subgraph.
-    subgraph_cache: Mutex<LruCache<DeploymentHash, SubgraphInfo>>,
+    subgraph_cache: Mutex<LruCache<(DeploymentHash, VersionNumber), SubgraphInfo>>,
```

Comment:
How about keeping `DeploymentHash` as the cache key and changing `SubgraphInfo.api` to be a `HashMap<Version, Arc<ApiSchema>>`? As long as there's only a few API versions, we could even populate that `HashMap` eagerly; once there are more versions, we might want to populate entries there on demand.

That would make it possible to not pass in a version to `subgraph_info` and only use it when we actually need an `ApiSchema` in the `QueryStore`

Reply:
eaa5c19a8384bd804bc8b25aa11489e3ba2dabc1